core/src/test/resources/fielderrorregion/Foo.java

```diff
@@ -0,0 +1,29 @@
+package test;
+
+import javax.annotation.Nullable;
+
+public class Foo {
+  Bar f0 = new Bar().process(null);
+  Bar f1;
+  Bar f2, f3;
+  @Nullable Bar nullableBar;
+  Bar f4 = nullableBar.process(new Object());
+
+  Foo() {
+    this.f2 = new Bar();
+  }
+
+  public void run1() {
```

Comment:
Why do we need both `run1` and `run2`?

Reply:
Yes exactly that was needed to produce more errors to make sure `@Nullable` is not inferred.

Regarding `f4`, I am not sure if I fully understood the question, making `f4` will resolve one error which is assigning `@Nullable` and triggers 2 new errors which is deref of `f4` therefore the actual effect should be `1`. I checked it manually the effect is 1 and correct. Please let me know if I am missing something.

Reply:
Ok, then, but, by that logic: shouldn't it be the same number for `f1` and `f3`? There is also one error being resolved (the missing initialization error) and two being added. But the corresponding `TReport`s have `2` as the effect. Either way, I don't understand the discrepancy between the number for `f4` and the other two.

Reply:
Actually the difference is that making method `f1` or `f3` individually will not resolve the initialization error. The error still remains after the injection. It needs both annotation to be inserted to resolve the error. It only creates two new errors and does not resolve any error, hence the effect is `+2`.

Reply:
What is the initialization error in question here?

Is it one error for `f1` not being initialized and one error for `f3` not being initialized, or is the generic error at `Foo()` saying some fields are not initialized by the constructor?

I thought we were serializing individual initialization errors. If we aren't, then wouldn't any class with 2 or more constructors and at least one field which is a) not initialized, b) treated as non-null in more than one place, then result in all non-initialized fields being marked `@SuppressWarnings(...)`? Even when many could otherwise be marked `@Nullable`? It seems like the auto-annotator should be able to consider each uninitialized field as a separate error...

To test this, could we have an example analogous to `f1` but without the explicit dereferences and make sure that is actually determined to be `@Nullable`?

Or am I missing a different reason why adding `@Nullable` to `f1` or `f3` doesn't remove the initialization error for that field?

Reply:
@lazaroclapp Actually we only serialize one error for initialization errors per constructor which can consist of multiple fields not being initialized. The main purpose of Annotator is to minimize the number of errors the user sees from NullAway output, therefore, we do not split an error coming from NullAway into multiple errors for each uninitialized field. However, the scenario which is mentioned does not happen as we serialize a fix object for each uninitialized field and nullability of each one is investigated individually. I added a unit test for that, please see 8fb982c6cb48a22c1958c6ae6cd2d6d86cafbc68

In this unit test, 

- `f1`, `f2` and `f3` are not initialized by constructors and are candidates for `@Nullable`. 
-  Marking `f1`, `f2` or `f3` as `@Nullable` individually will not resolve the one initialization error complaining about the other two fields.
- `f3` has both conditions, not initialized and treated as `@Nonnull` in more than one place.
- `f1` and `f2` are determined to be `@Nullable`.
- `f3` is determined to be `@Nonnull` and instead is annotated with `SuppressWarnings("NullAway.Init")`
- `f4` was not involved in initialization errors (just `assign nullable` error) and cannot be `@Nullable`, therefore it is marked with `@SuppressWarnings("NullAway")`.